### PR TITLE
Clippy Clean up

### PR DESCRIPTION
**Pull request by lthiery**
Clippy gets smarter! Also, nightly fixed

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions-rs/toolchain@v1
         with:
-          toolchain: nightly-2022-04-24
+          toolchain: nightly
```
Comment (lulf) [inline .github/workflows/rust.yml:57]: Using the latest nightly has a tendency to break unexpectedly, so I would recommend pinning just to a newer nightly version.

Comment (lthiery) [reply .github/workflows/rust.yml]: It actually looks like the async stack no longer requires nightly. Maybe we should just do an all-features build from stable and call it a day?

**Review by lulf — APPROVED** [review]

**Review by ivajloip — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 1 above)
Comment (ivajloip) [reply .github/workflows/rust.yml]: If we can switch to stable, that would be great I think. :)

Comment (ivajloip) [reply .github/workflows/rust.yml]: If we can switch to stable, that would be great I think. :)

Comment (lulf) [reply .github/workflows/rust.yml]: Good point, if it works with stable that is better.

**Pull request merged**